File: antlr/src/main/java/com/rescribe/antlr/parse/schema/Function.java
package com.rescribe.antlr.parse.schema;

import lombok.*;
import lombok.experimental.SuperBuilder;

@AllArgsConstructor
@NoArgsConstructor
@Data
@SuperBuilder
@EqualsAndHashCode(callSuper = true)
public class Function extends NestedObject {
  String name;
  String returnType;
  boolean isconstructor;
  // String content;
}

File: api/src/elastic/antlrMappings/function.ts
import nestedMappings from './nestedObject';

const functionMappings = {
  ...nestedMappings,
  name: {
    type: 'text',
    analyzer: 'trigrams'
  },
  returnType: {
    type: 'text'
  },
  isConstructor: {
    type: 'boolean'
  }
};

export default functionMappings;
